packages/activity/src/index.ts

```diff
@@ -289,6 +298,11 @@ export class Context {
    */
   public readonly metricMeter: MetricMeter;
 
+  /**
+   * Holder object for activity cancellation details
+   */
+  public readonly cancellationDetails: ActivityCancellationDetailsHolder;
```

Comment:
I understand why you introduced this holder. I'm not convinced this is the best solution. An alternative would have been to split `Context` into an interface AND an implementation class. We'd obviously need to keep to export a `Context` object, to keep the `current()` static method, but that wouldn't be hard to do. But whatever, there's nothing wrong with the holder approach.

My real concern is that we're making that indirection be part of the public `Context` API. That's not desirable from an API perspective, and totally avoidable by keeping that field private, and exposing a `cancellationDetails` property getter instead.